File: 백준/Silver/1764. 듣보잡/듣보잡.py
import sys

N, M = map(int, sys.stdin.readline().strip().split())
human = dict()
result = []
for i in range(N):
  human[sys.stdin.readline().strip()] = 1

for i in range(M):
  name = sys.stdin.readline().strip()
  temp = human.get(name)
  if temp == None:
    human[temp] = 1
  elif temp == 1:
    result.append(name)
    
result.sort()
print(len(result))
for i in range(len(result)):
  print(result[i])

File: 백준/Silver/1764. 듣보잡/듣보잡.java

import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

		List<String> list = new ArrayList<>();
		// 듣도 못한 사람 set
		Set<String> set = new HashSet<>();


		String[] sa = bf.readLine().split(" ");
		int N = Integer.parseInt(sa[0]);
		int M = Integer.parseInt(sa[1]);

		for(int i=0; i<N; i++){
			set.add(bf.readLine());
		}

		for(int i=0; i<M; i++){
			String name = bf.readLine();
			if(set.contains(name)){
				list.add(name);
			}
		}

		System.out.println(list.size());
		StringBuilder sb = new StringBuilder();
		Collections.sort(list);
		for(int i=0; i<list.size(); i++){
			sb.append(list.get(i)).append("\n");
		}

		System.out.println(sb);

	}
}
